cmd/minishift/cmd/setup.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright (C) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/minishift/minishift/pkg/minishift/setup/hypervisor"
+	"github.com/minishift/minishift/pkg/util/os"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
+)
+
+const (
+	assumeYesFlag = "yes"
+)
+
+var (
+	assumeYes bool
+)
+
+// setupCmd represents the command to setup Minishift pre-requisites
+var setupCmd = &cobra.Command{
+	Use:   "setup",
+	Short: "Configures pre-requisites for Minishift on the host machine",
+	Long:  `Configures pre-requisites for Minishift on the host machine`,
+	Run:   runSetup,
+}
+
+func init() {
+	setupCmd.Flags().BoolVarP(&assumeYes, assumeYesFlag, "y", false, "Automatically consider 'yes' for all questions")
+	RootCmd.AddCommand(setupCmd)
+}
+
+func runSetup(cmd *cobra.Command, args []string) {
+	if os.CurrentOS() == "windows" {
+		if err := hypervisor.CheckHypervisorAvailable(); err != nil {
+			atexit.ExitWithMessage(1, err.Error())
+		}
+
+		if err := hypervisor.CheckAndConfigureHypervisor(); err != nil {
+			atexit.ExitWithMessage(1, err.Error())
+		}
+
+		fmt.Println("Pre-requisites are ready.\nYou can now proceed with 'minishift start'.")
+	} else {
+		fmt.Println("The implementation of this command is not available yet in this OS.\nIt will be available in future version.\nPlease continue with setting pre-requisites through manual process.")
```

Comment:
We should add `https://docs.okd.io/latest/minishift/getting-started/index.html` to this message. 
s/The implementation of this command is not available yet in this OS.\nIt will be available in future version/`The implementation of this command is not available for this operating system.`/